contrib/cli/great_expectations_contrib/commands.py

```diff
@@ -0,0 +1,135 @@
+import os
+import subprocess
+import sys
+from collections import namedtuple
+
+import click
+from cookiecutter.main import cookiecutter
+
+Command = namedtuple("Command", ["name", "args", "error_message"])
+
+
+def init_cmd(url: str) -> None:
+    """
+    Initializes a contributor package by pulling down the Cookiecutter template
+    and hydrating it.
+    """
+    echo("Configure your template:\n", "blue", bold=True)
+    cookiecutter(url, overwrite_if_exists=False)
+    echo("\nSuccessfully set up contrib package!", "green", bold=True)
+
+
+def check_cmd() -> None:
+    """
+    Performs a series of checks on a contributor package.
+    These include code style, testing, docstrings, and more.
+    """
+    perform_check(suppress_output=False)
+
+
+def publish_cmd() -> None:
+    """
+    Performs same checks as `check_cmd`; if they pass, the user is prompted to
+    supply PyPi credentials. Valid inputs will result in an uploaded package.
+    """
+    success = perform_check(suppress_output=True)
+    if not success:
+        echo(
+            "Please run the `check` command to diagnose before publishing",
+            "red",
+            bold=True,
+        )
+        return
+
+    echo("All checks have succeeded; you are ready to publish!", "green", bold=True)
+    publish_to_pypi()
+
+
+def perform_check(suppress_output: bool) -> bool:
+    commands = [
+        Command(
+            "black",
+            "black --check .",
+            "Please ensure that your files are linted properly with `black .`",
+        ),
+        Command(
+            "isort",
+            "isort --profile black --check .",
+            "Please ensure that your imports are sorted properly with `isort --profile black .`",
+        ),
+        Command(
+            "pytest",
+            "pytest .",
+            "Please ensure that you've written tests and that they all pass",
+        ),
+        Command(
+            "mypy",
+            "mypy --ignore-missing-imports --disallow-untyped-defs --show-error-codes --exclude venv .",
+            "Please ensure that all functions are type hinted",
+        ),
+    ]
+
+    successes = 0
+    for i, command in enumerate(commands):
```

Comment:
If you want to keep the function signature of `run_command` as it is (accepting an idx arg), you can tell `enumerate` to start counting at 1 (i.e. `enumerate(commands, 1)`) so you don't have to do `i + 1`

Reply:
I'm going to remove the index entirely based on your point below!